front50-proto/src/main/proto/applications.proto

```diff
@@ -0,0 +1,241 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+package spinnaker.front50.v3;
+
+option java_package = "com.netflix.spinnaker.front50.proto";
+option java_multiple_files = true;
+
+// ApplicationService is the service for interacting with the Spinnaker application domain.
+service ApplicationService {
+  // ListApplications provides a way of searching through Applications..
+  rpc ListApplications (ListApplicationsRequest) returns (ListApplicationsResponse);
+
+  // GetApplication returns a single Application matching the criteria provided.
+  rpc GetApplication (GetApplicationRequest) returns (Application);
+
+  // CreateApplication will create a new Application. If an Application already exists by its name, this operation
+  // will fail.
+  rpc CreateApplication (CreateApplicationRequest) returns (Application);
+
+  // UpdateApplication will update an existing Application. If an Application does not exist at the provided name, this
+  // operation will fail.
+  rpc UpdateApplication (UpdateApplicationRequest) returns (Application);
+
+  // DeleteApplication will delete an Application. This operation is unaffected by prior state.
+  rpc DeleteApplication (DeleteApplicationRequest) returns (google.protobuf.Empty);
+
+  // ListApplicationRevisions returns an Application's revision history.
+  rpc ListApplicationRevisions (ListApplicationRevisionsRequest) returns (ListApplicationRevisionsResponse);
+}
+
+// An Application includes all configuration for a particular service within Spinnaker, and is an aggregate of
+// associated infrastructure collections such as clusters, server groups, firewalls, load balancers, etc.
+//
+// You will typically create a different application for each service, though Spinnaker does not enforce that.
+message Application {
+  // Required. Resource name for the application (e.g. "applications/front50")
+  string name = 1;
+
+  // Required. The official name of the application (e.g. "front50")
+  string title = 2;
+
+  // The description of the application's purpose and function.
+  string description = 3;
+
+  // Required. The email address (typically a distribution list) of the owner to the application.
+  string owner_email = 4;
```

Comment:
We have some cases of joint ownership, where this becomes `repeated`.

Reply:
We have some orgs that have a similar setup (like in the case of transitioning ownership in reorgs, etc). I wonder if having something like `repeated Owner(string email, string label, bool primary)` might make more sense? It feels weird to me to have more than 1 owner. Interested to hear your thoughts: In practice, would additional metadata (like a label, primary flag, etc) reduce any confusion about ownership / who someone should contact for a particular context?